backend/app/Http/Controllers/StackController.php

```diff
@@ -12,20 +12,31 @@ class StackController extends Controller
 {
     public function addStack(StackRequest $request): JsonResponse
     {
-        $data = $request->all();
+        $name = $request->name;
+        $user_id = $request->user["id"];
+        $data = [
+            'name' => $name,
+            'user_id' => $user_id
+        ];
         try {
-            if($data){
-                Stack::create($data);
-                return $this->sendResponse(false, null, 'Stack created', $data, Response::HTTP_CREATED);
-            }else {
-                return $this->sendResponse(true, 'Stack failed', Response::HTTP_BAD_REQUEST);
+            if(empty($name)){
+                return $this->sendResponse(true, 'stack name is required', 'missing stack name',Response::HTTP_BAD_REQUEST );
+            }
+            $stack = Stack::where('name', $name);
+
+            if ($stack->count() > 0) {
+                return $this->sendResponse(true, "This stack already exists", "Duplicate stack name.",
+                    null, 404);
```

Comment:
pls use the standard http reponse.

Reply:
Please what is the standard HTTP response?

Reply:
look at what you wrote on line 23 Response::HTTP_BAD_REQUEST.. so follow that standard.
Please close one of your PRs it seems they are the same.

Reply:
> look at what you wrote on line 23 Response::HTTP_BAD_REQUEST.. so follow that standard.

Okay. Thank you.


>Please close one of your PRs it seems they are the same.

One is for addStack, the other is to update stack. They are different